src/Infrastructure/ErrorLoggingFileFetcher.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Infrastructure;
+
+use FileFetcher\FileFetcher;
+use FileFetcher\FileFetchingException;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+
+class ErrorLoggingFileFetcher implements FileFetcher, LoggerAwareInterface {
+
+	private $wrappedFileFetcher;
+
+	private $logger;
+
+	public function __construct( FileFetcher $fileFetcher, LoggerInterface $logger ) {
+		$this->wrappedFileFetcher = $fileFetcher;
+		$this->logger = $logger;
+	}
+
+	public function fetchFile( $fileUrl ) {
+		try {
+			return $this->wrappedFileFetcher->fetchFile( $fileUrl );
+		} catch ( FileFetchingException $e ) {
+			$this->logger->error( $e->getMessage(), [
+				'code' => $e->getCode(),
+				'trace' => $e->getTraceAsString()
+			] );
+			return '';
```

Comment:
I am not too sure about returning an empty string. Like this it's an ErrorLoggingAndSupressingFileFecther. Is there a reason to not rethrow the exception? Perhaps the user of the FileFetcher assumes an exception is thrown on error, as per the interface, and will then do the wrong thing when it gets an empty string? Or perhaps it will assume the error is getting suppressed and then fails when it is not. Like this it falls to the code constructing the object graph to make sure everything fits together, and to do so it needs to know about details of the implementations, "peeking behind" the interface, which I find very suspicious. Perhaps there is an issue with the interface itself?